src/org/stringtemplate/v4/STGroup.java

```diff
@@ -80,9 +80,11 @@
 		TEMPLATE_FILE_EXTENSION = ".st";
 	}
 
-    /** When we use key as a value in a dictionary, this is how we signify. */
-    public static final String DICT_KEY = "key";
-    public static final String DEFAULT_KEY = "default";
+    /** When we use key as a value in a dictionary, this is how we signify.
+     *  Magic suffix is used to distinguish "default" string literal and "default" key
+     *  End part is the date of this suffix adding. */
+    public static final String DICT_KEY = "key_st4_magic_suffix_2016/01/10";
+    public static final String DEFAULT_KEY = "default_st4_magic_suffix_2016/01/10";
```

Comment:
:exclamation: The string you use here is *less likely* to cause a problem, but all it does is change the magic string which breaks dictionary functionality. Use the following instead:

```java
public static final String DEFAULT_KEY = "\0";
```

This single character string is the shortest (and thus fastest to look up in a map) string which is not representable in the StringTemplate language for dictionaries.